src/sparse/node.rs

```diff
@@ -412,30 +424,38 @@ where
     StorageType: StorageInspect<NodesTable>,
     StorageType::Error: fmt::Debug,
 {
-    pub fn left_child(&self) -> Option<Self> {
+    pub fn left_child(&self) -> Result<Option<Self>, DeserializeError> {
         assert!(self.is_node());
         let key = self.node.left_child_key();
         if key == zero_sum() {
-            return Some(Self::new(self.storage, Node::create_placeholder()));
+            return Ok(Some(Self::new(self.storage, Node::create_placeholder())));
         }
         let buffer = self.storage.get(key).unwrap();
-        buffer.map(|b| {
-            let node = Node::from_buffer(*b);
-            Self::new(self.storage, node)
-        })
+        if let Some(buffer) = buffer {
+            buffer
+                .into_owned()
+                .try_into()
+                .map(|node: Node| Some(Self::new(self.storage, node)))
+        } else {
+            Ok(None)
+        }
```

Comment:
```suggestion
        buffer.map(|buffer| buffer.into_owned().try_into().map(|node| Self::new(self.storage, node))).transpose()
```

Reply:
Thanks for the suggestion. I think more concise code is better, but I am not sure if I find that the nested `map` statements make it more readable. 
After formatting, the code looks like this:
```rust
        buffer
            .map(|buffer| {
                buffer
                    .into_owned()
                    .try_into()
                    .map(|node| Self::new(self.storage, node))
            })
            .transpose()          
```
which still feels a little complex to me.
I will think about this one!

Reply:
What do you think about this:
```rust
...
        Ok(buffer
            .map(|buffer| buffer.into_owned().try_into())
            .transpose()?
            .map(|node| Self::new(self.storage, node)))
```